sql/hive/src/test/scala/org/apache/spark/sql/sources/JsonHadoopFsRelationSuite.scala

```diff
@@ -28,6 +28,14 @@ import org.apache.spark.sql.types._
 class JsonHadoopFsRelationSuite extends HadoopFsRelationTest {
   override val dataSourceName: String = "json"
 
+  // JSON does not write data of NullType and does not play well with BinaryType.
+  override protected def supportsDataType(dataType: DataType): Boolean = dataType match {
+    case a: NullType => false
+    case b: BinaryType => false
```

Comment:
But we do handle `BinaryType` [in `JacksonGenerator`] [1]. Does this mean we can save a DataFrame containing `BinaryType` field(s) to JSON files via `df.toJSON.saveAsTextFile(...)`, but can't read it back with `JSONRelation`?

[1]: https://github.com/apache/spark/pull/8597/files#diff-0f04c36e499d4dcf6931fbd62b3aa012R101